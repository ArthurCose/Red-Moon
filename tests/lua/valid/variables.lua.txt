-- saved as txt to ignore warnings

print("assignment:")
a = 1
b, c = 2, 3
d = 4, 5, 6, 7, 8
e, f = 9 -- testing clearing out old results (the line above)
print(a, b, c, d, e, f)

print("\nlocals:")
local b = 12
do
  local a
  print(a)
  a = 9
  local b, c = 8, 7
  print(a, b, c)
  do
    local a, b, c = 3, 4
    print(a, b, c)
  end
  print(a, b, c)
end
print(a, b, c)
print(_ENV.a, _ENV.b, _ENV.c)

print("\nnested exp_list:")
local foo = function(...) return ... end
local bar = function(a, b, c) return c end
print(bar(foo(1, 2, 3)), 0)

print("\noperations with variable return:")
print(false or foo(1, 2, 3))
print(1 + foo(1, 2, 3))
print(foo(1, 2, 3) + 1)

print("\nempty variadic:")
local empty_variadic = function(...) print(1, ...) end
empty_variadic()

print("\nexpression assignment:")
local a = {}
a[1] = 4
function foo() return a end
foo()[2] = 5
(a)[3] = 6
print(a[1], a[2], a[3])
