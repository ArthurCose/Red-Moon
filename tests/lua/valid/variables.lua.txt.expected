assignment:
1	2	3	4	9	nil

locals:
nil
9	8	7
3	4	nil
9	8	7
1	12	3
1	2	3

shadowing:
6

nested exp_list:
3	0

operations with variable return:
1
2
2

empty variadic:
1

expression assignment:
4	5	6
